module_4/ex_4_6_5.py

```diff
@@ -0,0 +1,152 @@
+'''
+Consignas:
+    Escribir funciones que resuelvan los siguientes problemas:
+    a) Dado un año indicar si es bisiesto.
+    Nota: un año es bisiesto si es un número divisible por 4, pero no si es divisible por 100, excepto que también sea divisible por 400.
+    b) Dado un mes y un año, devolver la cantidad de días correspondientes.
+    c) Dada una fecha (día, mes, año), indicar si es válida o no.
+    d) Dada una fecha, indicar los días que faltan hasta fin de mes.
+    e) Dada una fecha, indicar los días que faltan hasta fin de año.
+    f) Dada una fecha, indicar la cantidad de días transcurridos en ese año hasta esa fecha.
+    g) Dadas dos fechas (día1, mes1, año1, día2, mes2, año2), indicar el tiempo transcurrido entre ambas, en años, meses y días.
+    Nota: en todos los casos, invocar las funciones escritas previamente cuando sea posible.
+'''
+DIC_MESES_DIAS = {0:0, 1:31, 2:28, 3:31, 4:30, 5:31, 6:30, 7:31, 8:31, 9:30, 10:31, 11:30, 12:31}
+
+def es_bisiesto(y):
+    '''
+        Función que retorna True o False si un año es o no bisiesto.
+    '''
+    if(y % 400 == 0) or (y % 4 == 0 and y % 100 != 0):
```

Comment:
los parentesis no son necesarios. siempre al momento de evaluar de manera logica se van a evaluar primeros los and y posterior los or. por lo que esto haria primero el `y % 400 == 0` y en caso de que sea falsa la condicion haria el `y % 4 == 0 and y % 100 != 0`.

al momento de dividir en terminos por asi decirlo es primero el and que funcionaria como la multiplicacion y el or que seria como la suma.

4 * 4 + 8 -> para resolver esto resolvemos primero la multiplicacion y luego la suma. de forma logica con and y or hacemos lo mismo. 4 and 4 or 8. no resolveriamos jamas el or antes que los and.

por ende esto podria escribirse tranquilamente sin los parentesis. ojo, si te es mas facil de entender esta perfecto dejarlos. pero no esta de mas saber como funciona esta evaluacion logica por si lo ves algun dia sin parentesis

Reply:
esta division de terminos NO significa que si tengo en java una sentencia como la siguiente vaya a ejecutar el and primero:
objeto == null || objeto.isEnabled() && isValid

siempre parte por los OR evalua la primera condicion y si no se cumple evalua la segunda y recien en la segunda va a evaluar los and. esto pasa tanto en python como en java como en c como everywhere